### PR TITLE
Add font support

### PR DESCRIPTION
**Pull request by Rorschach1219**
Configurable JSON file, supporting local font.  I tested Chinese, Japanese and Korean and worked well.

汉字
韩国
日文
[fonts.zip](https://github.com/yamashi/CyberEngineTweaks/files/5832003/fonts.zip)

### CODE REVIEW

File: src/console/Console.cpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ void Console::Update()
     ImGui::SetNextWindowPos(ImVec2(0.f, 0.f), ImGuiCond_FirstUseEver);
     ImGui::SetNextWindowSize(ImVec2(resolution.cx, resolution.cy * 0.3f), ImGuiCond_FirstUseEver);
 
-    ImGui::Begin("Cyber Engine Tweaks");
```
Comment (yamashi) [inline src/console/Console.cpp:46]: We are not going to add parenthesis for everything we add

File: src/d3d12/D3D12_Functions.cpp [hunk 2]
```diff
@@ -318,6 +319,39 @@ bool D3D12::InitializeImGui(size_t aBuffersCounts)
...
+        if (!options.FontPath.empty())
+        {
+            std::filesystem::path fontPath(options.FontPath);
+            if (!fontPath.is_absolute())
+            {
+                fontPath = options.CETPath / fontPath;
+            }
+            if (std::filesystem::exists(fontPath))
+            {
+                const ImWchar* glyph_ranges = io.Fonts->GetGlyphRangesDefault();
```
Comment (yamashi) [inline src/d3d12/D3D12_Functions.cpp:333]: cpGlyphRanges

File: src/d3d12/D3D12_Functions.cpp [hunk 3]
```diff
@@ -318,6 +319,39 @@ bool D3D12::InitializeImGui(size_t aBuffersCounts)
...
+                else if (options.FontGlyphRanges == "Korean")
+                    glyph_ranges = io.Fonts->GetGlyphRangesKorean();
+                else if (options.FontGlyphRanges == "Cyrillic")
+                    glyph_ranges = io.Fonts->GetGlyphRangesCyrillic();
+                else if (options.FontGlyphRanges == "Thai")
+                    glyph_ranges = io.Fonts->GetGlyphRangesThai();
+                else if (options.FontGlyphRanges == "Vietnamese")
+                    glyph_ranges = io.Fonts->GetGlyphRangesVietnamese();
+                ImFont* font = io.Fonts->AddFontFromFileTTF(fontPath.string().c_str(), options.FontSize, NULL, glyph_ranges);
+                if (font != NULL)
```
Comment (yamashi) [inline src/d3d12/D3D12_Functions.cpp:349]: nullptr

File: src/d3d12/D3D12_Functions.cpp [hunk 4]
```diff
@@ -318,6 +319,39 @@ bool D3D12::InitializeImGui(size_t aBuffersCounts)
...
+                    glyph_ranges = io.Fonts->GetGlyphRangesJapanese();
+                else if (options.FontGlyphRanges == "Korean")
+                    glyph_ranges = io.Fonts->GetGlyphRangesKorean();
+                else if (options.FontGlyphRanges == "Cyrillic")
+                    glyph_ranges = io.Fonts->GetGlyphRangesCyrillic();
+                else if (options.FontGlyphRanges == "Thai")
+                    glyph_ranges = io.Fonts->GetGlyphRangesThai();
+                else if (options.FontGlyphRanges == "Vietnamese")
+                    glyph_ranges = io.Fonts->GetGlyphRangesVietnamese();
+                ImFont* font = io.Fonts->AddFontFromFileTTF(fontPath.string().c_str(), options.FontSize, NULL, glyph_ranges);
```
Comment (yamashi) [inline src/d3d12/D3D12_Functions.cpp:348]: pFont, nullptr

**Review by yamashi — CHANGES_REQUESTED** [review]

**Comment by yamashi** [comment]
It seems that FontSize relies on having a font specified, can't we make it so FontSize works even if we don't pass a FontPath?

**Review by yamashi — APPROVED** [review]

**Comment by yamashi** [comment]
Thanks :)

**Comment by Rorschach1219** [comment]
The default font size is 13, which can be configured separately. can also set glyph ranges and font path at the same time
As shown below：

**Pull request merged**